type error in `.nuxt/i18n.options.mjs` with `checkJs: true` in `tsconfig.json`

### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.7.1
- CLI Version:      3.8.0
- Nitro Version:    2.6.3
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      modules
- Runtime Modules:  @nuxtjs/i18n@8.0.0-rc.4
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-xigg7k

run `npm run check` in the console of the above stackblitz repo

### Describe the bug

when `checkJs: true` is enabled in `tsconfig.json`, i get the following type errors when running `vue-tsc`:

```
.nuxt/i18n.options.mjs:1:14 - error TS7005: Variable 'localeCodes' implicitly has an 'any[]' type.

1 export const localeCodes = []
               ~~~~~~~~~~~

.nuxt/i18n.options.mjs:6:46 - error TS7006: Parameter 'context' implicitly has an 'any' type.

6 export const resolveNuxtI18nOptions = async (context) => {
                                               ~~~~~~~

.nuxt/i18n.options.mjs:12:37 - error TS7006: Parameter 'loader' implicitly has an 'any' type.

12  const vueI18nConfigLoader = async (loader) => {
                                       ~~~~~~

.nuxt/i18n.options.mjs:13:48 - error TS7006: Parameter 'r' implicitly has an 'any' type.

13             const config = await loader().then(r => r.default || r)
                                                  ~


Found 4 errors in the same file, starting at: .nuxt/i18n.options.mjs:1


```

### Additional context

_No response_

### Logs

_No response_

--- a/src/gen.ts
+++ b/src/gen.ts
@@ -100,7 +100,7 @@ export function generateLoaderOptions(
     })
   }
 
-  let genCode = ''
+  let genCode = '// @ts-nocheck\n'
   const localeInfo = options.localeInfo || []
   const syncLocaleFiles = new Set<LocaleInfo>()
   const asyncLocaleFiles = new Set<LocaleInfo>()

